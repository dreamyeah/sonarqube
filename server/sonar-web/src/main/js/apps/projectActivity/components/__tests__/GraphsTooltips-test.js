/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import React from 'react';
import { shallow } from 'enzyme';
import GraphsTooltips from '../GraphsTooltips';

const SERIES_OVERVIEW = [
  {
    name: 'code_smells',
    translatedName: 'Code Smells',
    style: 1,
    data: [
      {
        x: '2011-10-01T22:01:00.000Z',
        y: 18
      },
      {
        x: '2011-10-25T10:27:41.000Z',
        y: 15
      }
    ]
  },
  {
    name: 'bugs',
    translatedName: 'Bugs',
    style: 0,
    data: [
      {
        x: '2011-10-01T22:01:00.000Z',
        y: 3
      },
      {
        x: '2011-10-25T10:27:41.000Z',
        y: 0
      }
    ]
  },
  {
    name: 'vulnerabilities',
    translatedName: 'Vulnerabilities',
    style: 2,
    data: [
      {
        x: '2011-10-01T22:01:00.000Z',
        y: 0
      },
      {
        x: '2011-10-25T10:27:41.000Z',
        y: 1
      }
    ]
  }
];

const DEFAULT_PROPS = {
  formatValue: val => 'Formated.' + val,
  graph: 'overview',
  graphWidth: 500,
  measuresHistory: [],
  selectedDate: new Date('2011-10-01T22:01:00.000Z'),
  series: SERIES_OVERVIEW,
  tooltipIdx: 0,
  tooltipPos: 666
};

it('should render correctly for overview graphs', () => {
  expect(shallow(<GraphsTooltips {...DEFAULT_PROPS} />)).toMatchSnapshot();
});

it('should render correctly for random graphs', () => {
  expect(
    shallow(
      <GraphsTooltips
        {...DEFAULT_PROPS}
        graph="random"
        selectedDate={new Date('2011-10-25T10:27:41.000Z')}
        tooltipIdx={1}
      />
    )
  ).toMatchSnapshot();
});
