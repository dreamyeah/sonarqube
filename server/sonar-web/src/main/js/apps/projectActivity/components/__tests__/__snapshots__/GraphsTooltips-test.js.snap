// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly for overview graphs 1`] = `
<BubblePopup
  customClass="bubble-popup-right"
  position={
    Object {
      "left": 476,
      "top": 50,
      "width": 250,
    }
  }
>
  <div
    className="project-activity-graph-tooltip"
  >
    <div
      className="project-activity-graph-tooltip-title spacer-bottom"
    >
      <FormattedDate
        date={2011-10-01T22:01:00.000Z}
        format="LL"
      />
    </div>
    <table
      className="width-100"
    >
      <tbody>
        <GraphsTooltipsContentOverview
          measuresHistory={Array []}
          serie={
            Object {
              "data": Array [
                Object {
                  "x": "2011-10-01T22:01:00.000Z",
                  "y": 18,
                },
                Object {
                  "x": "2011-10-25T10:27:41.000Z",
                  "y": 15,
                },
              ],
              "name": "code_smells",
              "style": 1,
              "translatedName": "Code Smells",
            }
          }
          tooltipIdx={0}
          value="Formated.18"
        />
        <GraphsTooltipsContentOverview
          measuresHistory={Array []}
          serie={
            Object {
              "data": Array [
                Object {
                  "x": "2011-10-01T22:01:00.000Z",
                  "y": 3,
                },
                Object {
                  "x": "2011-10-25T10:27:41.000Z",
                  "y": 0,
                },
              ],
              "name": "bugs",
              "style": 0,
              "translatedName": "Bugs",
            }
          }
          tooltipIdx={0}
          value="Formated.3"
        />
        <GraphsTooltipsContentOverview
          measuresHistory={Array []}
          serie={
            Object {
              "data": Array [
                Object {
                  "x": "2011-10-01T22:01:00.000Z",
                  "y": 0,
                },
                Object {
                  "x": "2011-10-25T10:27:41.000Z",
                  "y": 1,
                },
              ],
              "name": "vulnerabilities",
              "style": 2,
              "translatedName": "Vulnerabilities",
            }
          }
          tooltipIdx={0}
          value="Formated.0"
        />
      </tbody>
    </table>
  </div>
</BubblePopup>
`;

exports[`should render correctly for random graphs 1`] = `
<BubblePopup
  customClass="bubble-popup-right"
  position={
    Object {
      "left": 476,
      "top": 50,
      "width": 250,
    }
  }
>
  <div
    className="project-activity-graph-tooltip"
  >
    <div
      className="project-activity-graph-tooltip-title spacer-bottom"
    >
      <FormattedDate
        date={2011-10-25T10:27:41.000Z}
        format="LL"
      />
    </div>
    <table
      className="width-100"
    >
      <tbody>
        <GraphsTooltipsContent
          serie={
            Object {
              "data": Array [
                Object {
                  "x": "2011-10-01T22:01:00.000Z",
                  "y": 18,
                },
                Object {
                  "x": "2011-10-25T10:27:41.000Z",
                  "y": 15,
                },
              ],
              "name": "code_smells",
              "style": 1,
              "translatedName": "Code Smells",
            }
          }
          value="Formated.15"
        />
        <GraphsTooltipsContent
          serie={
            Object {
              "data": Array [
                Object {
                  "x": "2011-10-01T22:01:00.000Z",
                  "y": 3,
                },
                Object {
                  "x": "2011-10-25T10:27:41.000Z",
                  "y": 0,
                },
              ],
              "name": "bugs",
              "style": 0,
              "translatedName": "Bugs",
            }
          }
          value="Formated.0"
        />
        <GraphsTooltipsContent
          serie={
            Object {
              "data": Array [
                Object {
                  "x": "2011-10-01T22:01:00.000Z",
                  "y": 0,
                },
                Object {
                  "x": "2011-10-25T10:27:41.000Z",
                  "y": 1,
                },
              ],
              "name": "vulnerabilities",
              "style": 2,
              "translatedName": "Vulnerabilities",
            }
          }
          value="Formated.1"
        />
      </tbody>
    </table>
  </div>
</BubblePopup>
`;
